pkg/roll/update_test.go

```diff
@@ -281,7 +297,7 @@ func SimulateRollingUpgradeDisable(t *testing.T, full, nonew bool) {
 		}
 
 		// calculate the next update
-		nextRemove, nextAdd := rollAlgorithm(len(old), len(new), target, target, minimum)
+		nextRemove, nextAdd := rollAlgorithm(len(old), len(new), target-len(new), len(new), target, minimum)
```

Comment:
the fuzz test is doing weird things to `oldDesired` because it may randomly choose to schedule on an old node or an empty node, which means `oldDesired != len(old)` sometimes... and also means that `nextRemove` is pretty much not being respected - it's not always the case that the number of nodes being removed will equal `nextRemove`